gems/aws-sdk-s3/spec/object/presigned_url_spec.rb

```diff
@@ -159,7 +159,7 @@ module S3
           allow(Time).to receive(:now).and_return(now)
           url, headers = obj.presigned_request(
             :get, expires_in: 86_400, request_payer: 'peccy')
-          expect(CGI.parse(url)).to eq(CGI.parse(
+          expect(URI.decode_www_form(url)).to eq(URI.decode_www_form(
```

Comment:
Are these actually equivalent? Is there any supporting documentation for these implementations?

Reply:
`CGI.parse`: https://docs.ruby-lang.org/en/3.4/CGI.html#method-c-parse
`URI.decode_www_form`: https://docs.ruby-lang.org/en/3.4/URI.html#method-c-decode_www_form

It treats it as `x-www-form-urlencoded`, which uses the query string format of `key=value&key2=value`.

They return data in a slightly different format. `CGI.parse` returns a hash, `URI.decode_www_form` an array of 2-element arrays. For tests to just check against this is fine, if it were exposed to users it would need a different solution.